app/scripts/store.js

```diff
@@ -328,6 +328,11 @@ var exports = (function() {
           this.state.activeModelContainer = undefined;
         }
 
+        // TODO: find better solution - now we do this to trigger an update on the front-end (vm.$forceUpdate() is added in Vue 2.0)
+        _.each(this.state.modelContainers, el => {
+          el.selected = false;
+        });
+
```

Comment:
This solution is not elegant, but vm.$forceUpdate() was added in Vue 2.0 for this reason: after a trash of a scenario you want to update a couple of components - but in Vue 1.0 you can't without diving into reactivity in depth. I did, but failed to trigger the update - this solution is sub-optimal, but at least the "selected models number" is correct (it will always be 0 after a Scenario trash).